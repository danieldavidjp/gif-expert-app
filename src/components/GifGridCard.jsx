import React from "react";
import PropTypes from "prop-types";

export const GifGridCard = ({ title, url }) => {
	return (
		<div className="card">
			<img src={url} alt={title} />
			<p>{title}</p>
		</div>
	);
};

GifGridCard.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};
