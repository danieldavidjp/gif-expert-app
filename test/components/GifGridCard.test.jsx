import { render, screen } from "@testing-library/react";
import { GifGridCard } from "../../src/components/GifGridCard";

describe("Tests for GifGridCard component", () => {
	const title = "Harry Potter";
	const url = `https://media.giphy.com/media/mz1kJeDVueKC4/giphy.gif`;

	test("should match the snapshot", () => {
		const { container } = render(<GifGridCard title={title} url={url} />);

		expect(container).toMatchSnapshot();
	});

	test("must show the image with the url and alt indicate", () => {
		render(<GifGridCard title={title} url={url}></GifGridCard>);
		// screen.debug();

		// expect(screen.getByRole('img').src).toBe(url);
		// expect(screen.getByRole('img').alt).toBe(title);
		const { src, alt } = screen.getByRole("img");
		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test("must show the title of the component", () => {
		render(<GifGridCard title={title} url={url} />);

		expect(screen.getByText(title)).toBeTruthy();
	});
});
