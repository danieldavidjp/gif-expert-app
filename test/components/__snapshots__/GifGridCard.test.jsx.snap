// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifGridCard component should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Harry Potter"
      src="https://media.giphy.com/media/mz1kJeDVueKC4/giphy.gif"
    />
    <p>
      Harry Potter
    </p>
  </div>
</div>
`;
